sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -150,6 +150,16 @@ object SQLConf {
     }
   }
 
+  val DETERMINISTIC_UDF_FOLD_ENABLED = buildConf("spark.deterministic.udf.folding.enabled")
```

Comment:
- We need `spark.sql` prefix. 
- Let's move this new configuration to the end of the other configurations. Maybe, after `DEFAULT_V2_CATALOG`.

Reply:
The name is misleading. We don't folding all deterministic UDF, do we? This PR only works for the literals.

Reply:
I will change it to have spark.sql in the conf name. Do you prefer the conf name to be spark.sql.deterministic.literal.udf.folding.enabled?  That seems long..  I'm open to suggestions for the name.  Thanks.

Reply:
I didn't hear back, so I used the name I suggested here in the latest push. Let me know if you have any other comments.